### PR TITLE
Add JSON config

### LINKED ISSUE #2: Config: JSON-based
We start with a simple file-based JSON configurations.

We'll use `serde_json` parsing/encoding.

### PR DESCRIPTION
**Pull request by amackillop**
Adds a json config for the hack server. It will load the config from the storage dir path. For now the file is called `config.json`. 

Closes #2

### CODE REVIEW

File: src/config.rs [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+			"debug" => LogLevel::Debug,
+			"info" => LogLevel::Info,
+			"warn" => LogLevel::Warn,
+			"error" => LogLevel::Error,
+			_ => return Err(anyhow!(
+				"Unsupported log level: {}. Use one of [gossip, trace, debug, info, warn, error] ",
+				json_config.log_level
+			)),
+		};
+		let config = Config {
```
[the remark this answers is not in the document]
Comment (amackillop) [reply src/config.rs:36]: sure

File: src/config.rs [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+pub struct Config {
+	pub esplora_server_url: String,
+	pub listening_addr: SocketAddress,
+	pub log_level: LogLevel,
+	pub network: Network,
+}
+
+impl Config {
+	pub fn new<P: AsRef<Path>>(storage_dir_path: P) -> Result<Self> {
+        let contents = std::fs::read_to_string(storage_dir_path.as_ref().join(CONFIG_FILE_NAME))?;
```
[the remark this answers is not in the document]
Comment (amackillop) [reply src/config.rs:20]: Like a utils::read_config(..) function?

[the remark this answers is not in the document]
Comment (amackillop) [reply src/config.rs]: Done

File: src/config.rs [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+#[derive(PartialEq, Eq, Debug)]
+pub struct Config {
+	pub esplora_server_url: String,
+	pub listening_addr: SocketAddress,
+	pub log_level: LogLevel,
+	pub network: Network,
+}
+
+impl Config {
+	pub fn new<P: AsRef<Path>>(storage_dir_path: P) -> Result<Self> {
```
[the remark this answers is not in the document]
Comment (amackillop) [reply src/config.rs]: It's done that way in the utility function

**Review by saidalaoui-sq — APPROVED** [review]
Pending `cargo fmt` fix to get this merged

**Comment by amackillop** [comment]
Rebased

File: src/config.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (amackillop) [reply src/config.rs]: Yeah I could do that it's a bit simpler

**Pull request merged**